tests/pipelines/test_pipelines.py

```diff
@@ -2175,3 +2176,155 @@ def test_ddpm_ddim_equality_batched(self):
 
         # the values aren't exactly equal, but the images look the same visually
         assert np.abs(ddpm_images - ddim_images).max() < 1e-1
+
+
+class TestLoraHotSwapping(unittest.TestCase):
```

Comment:
Thanks for the test! I think it's checking recompilation on an individual model-level, which is good. 

However, to have a more realistic scenario covered, that is aligned with the users, I think we will need to have a full pipeline example, too.

Reply:
Since we're only testing at the model-level here, would prefer this to live in `tests/models/test_modeling_common.py`. We could try generalizing the test similar to `test_save_load_lora_adapter()` so that we know this is supported across all the models that are a subclass of `PeftAdapterMixin`. WDYT?